voicebio/internal/client/client.go

```diff
@@ -0,0 +1,325 @@
+
+	"google.golang.org/grpc"
+)
+
+const defaultStreamingBufsize uint32 = 1024
+
+type Client struct {
+	voicebio         voicebiopb.VoiceBioServiceClient
+	conn             *grpc.ClientConn
+	log              log.Logger
+	streamingBufSize uint32
+}
+
+func NewClient(conn *grpc.ClientConn, opts ...Option) (*Client, error) {
+	var args clientArgs
+
+	args.streamingBufSize = defaultStreamingBufsize
+
+	for _, opt := range opts {
+		err := opt(&args)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create a client: %v", err)
+		}
+	}
+
+	if args.log == nil {
+		args.log = log.NewDiscardLogger()
+	}
+
+	return &Client{
+		voicebio:         voicebiopb.NewVoiceBioServiceClient(conn),
+		conn:             conn,
+		streamingBufSize: args.streamingBufSize,
+		log:              args.log,
+	}, nil
+}
+
+type clientArgs struct {
+	log              log.Logger
+	streamingBufSize uint32
+}
+
+// Option configures how we setup the connection with a server.
+type Option func(*clientArgs) error
+
+// WithStreamingBufferSize returns an Option that sets up the buffer size
+// (bytes) of each message sent from the Client to the server during streaming
+// GRPC calls.  Use this only if Cobalt recommends you to do so.  A value n>0 is
+// required.
+func WithStreamingBufferSize(n uint32) Option {
+	return func(c *clientArgs) error {
+		if n == 0 {
+			return fmt.Errorf("invalid streaming buffer size of 0")
+		}
+
+		c.streamingBufSize = n
+
+		return nil
+	}
+}
+
+func WithLogger(logger log.Logger) Option {
+	return func(c *clientArgs) error {
+		c.log = logger
+		return nil
+	}
+}
+
+func (c *Client) CobaltVersions(ctx context.Context) (string, error) {
+	v, err := c.voicebio.Version(ctx, &voicebiopb.VersionRequest{})
+	if err != nil {
+		return "", err
+	}
+
+	return v.Version, nil
+}
+
+func (c *Client) ListModels(ctx context.Context) ([]*voicebiopb.Model, error) {
+	resp, err := c.voicebio.ListModels(ctx, &voicebiopb.ListModelsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Models, nil
+}
+
+func (c *Client) StreamingEnroll(ctx context.Context,
+	cfg voicebiopb.EnrollmentConfig, //nolint:govet // cfg is a large struct but we want to use a copy
+	audio io.Reader) (*voicebiopb.StreamingEnrollResponse, error) {
+
+	stream, err := c.voicebio.StreamingEnroll(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := sendaudioEnroll(stream, &cfg, audio, c.streamingBufSize)
+	if err != nil && err != io.EOF {
+		// if sendaudio encountered io.EOF, it's only a
+		// notification that the stream has closed.  The actual
+		// status will be obtained in a subsequent Recv call, in
+		// the other goroutine below.  We therefore only forward
+		// non-EOF errors.
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+// sendaudio sends audio to a stream.
+func sendaudioEnroll(stream voicebiopb.VoiceBioService_StreamingEnrollClient,
+	cfg *voicebiopb.EnrollmentConfig, audio io.Reader,
+	bufsize uint32) (*voicebiopb.StreamingEnrollResponse, error) {
```

Comment:
This may be a good candidate to turn into a generic function sort of like what we do on the [server](https://github.com/cobaltspeech/voicebio/blob/fb38e4f8e203ef4d21ad1dd63bfe762bb9112d4f/internal/server/grpc/v1/streaming.go#L136), but not necessary for this PR

```suggestion
func sendAudio[TStream Any, TCfg Any, (stream voicebiopb.VoiceBioService_StreamingEnrollClient,
	cfg *voicebiopb.EnrollmentConfig, audio io.Reader,
	bufsize uint32) (*voicebiopb.StreamingEnrollResponse, error) {
```